lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -390,9 +388,71 @@ def set_information(self, trans, id, payload={}, **kwd):
             trans.sa_session.add(user_address)
         trans.sa_session.add(user)
         trans.sa_session.flush()
+        if trans.app.config.user_activation_on:
+            if self.send_verification_email(trans, user.email, user.username):
+                message = 'The login information has been updated with the changes.<br>Verification email has been sent to your new email address. Please verify it by clicking the activation link in the email.<br>Please check your spam/trash folder in case you cannot find the message.'
+            else:
+                message = 'Unable to send activation email, please contact your local Galaxy administrator.'
+                if trans.app.config.error_email_to is not None:
+                    message += ' Contact: %s' % trans.app.config.error_email_to
+                raise MessageException(message)
```

Comment:
@nsoranzo But then you will be sending to email that has not been saved to DB and there is a lot of code in between these two points that can fail I think.